server/src/websocket/handlers/activationKgStatus.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright 2022 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Channel } from "../index";
+import { WsMessage } from "../WsMessages";
+import config from "../../config";
+import logger from "../../logger";
+import fetch from "cross-fetch";
+import * as util from "util";
+
+type ActivationStatus = {
+  [key: number]: number
+}
+
+function sendMessage(data: string, channel: Channel) {
+  const info = new WsMessage({ message: data }, "user", "activation");
+  channel.sockets.forEach(socket => socket.send(info.toString()));
+}
+
+function getActivationStatus(projectIds: number[], channel: Channel): void {
+  const { gatewayUrl } = config.deployment;
+  for (let i = 0; i < projectIds.length; i++) {
+    const id = projectIds[i];
+    const activationStatusURL = `${gatewayUrl}/projects/${id}/graph/status`;
+    logger.info(`Fetching activation from ${id} projects`);
+
+    fetch(activationStatusURL).then(async (response) => {
+      const status = await response.json();
+      const previousStatuses = channel.data.get("activationProjects") as ActivationStatus;
+      const currentStatus = status?.progress ?? -1;
+      const previousStatus = previousStatuses ? previousStatuses[id] : null;
+      if (!util.isDeepStrictEqual(previousStatus, currentStatus)) {
+        const currentStatuses = { ...previousStatuses, [id]: currentStatus };
+        sendMessage(JSON.stringify(currentStatuses), channel);
+        channel.data.set("activationProjects", currentStatuses);
```

Comment:
```suggestion
        channel.data.set("activationProjects", currentStatuses);
        // eliminate possible duplicates
        channel.data.set("projectsIds", [...new Set(ids)]);
```
Duplicates may happen (not very likely, but maybe some latency and a repeated click from the user..).
Passing the array to a Set will prevent it.